modules/buckets/outputs.tf

```diff
@@ -1,9 +1,37 @@
 ##############################################################################
 # Outputs
 ##############################################################################
+# expand buckets, pull all elements except bucket_resource_key
+# iterate over list (...) and merge() back to map
 output "buckets" {
   description = "Map of buckets created in the Cloud Object Storage Instance"
-  value       = module.buckets
+  value = merge([for name, bucket in module.buckets : { (name) = {
```

Comment:
Presumably this is a different format output. I think this makes it a breaking change as any consumers of this output will need code change to adopt the updated bucket module... @ocofaigh

Reply:
A breaking change is only when infrastructure gets destroyed. Changed in outputs in not a breaking change. See https://terraform-ibm-modules.github.io/documentation/#/versioning

Reply:
It is not a change.

The bucket_resource_key is a new output of the root module.
This code de-constructs the output of the root module and re-constructs it without the new key.
So the output is the same. I applied the `main` branch and changed to `resource-key-output` branch and terraform plan/apply showed the output `buckets` was not changed.